data_structures/src/transaction_factory.rs

```diff
@@ -122,12 +123,14 @@ pub trait OutputsCollection {
 
     /// Generic inputs/outputs builder: can be used to build
     /// value transfer transactions and data request transactions.
+    #[allow(clippy::cast_sign_loss)]
+    #[allow(clippy::cast_possible_truncation)]
     #[allow(clippy::too_many_arguments)]
     fn build_inputs_outputs(
         &mut self,
         outputs: Vec<ValueTransferOutput>,
         dr_output: Option<&DataRequestOutput>,
-        fee: u64,
+        fee: f64,
         fee_type: FeeType,
```

Comment:
An alternative to stop clippy from complaining about casts, is to use a single enum `Fee` instead of `(f64, FeeType)`, like:

```rust
enum Fee {
    Absolute(u64),
    Weighted(f64),
}
```

Not sure what to do with the wallet API, I imagine that javascript code already handles fees as floats so we don't gain much with a change like this.

Reply:
Great idea. This is implemented now, but I'll need to fix some tests as we're essentially changing things up in the Wallet API (I'll try to make those backwards-compatible as much as possible)